src/diffusers/models/modeling_utils.py

```diff
@@ -758,89 +850,106 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
+                    user_agent=user_agent,
+                    commit_hash=commit_hash,
                 )
-                # TODO: https://github.com/huggingface/diffusers/issues/10013
-                if hf_quantizer is not None or dduf_entries:
-                    model_file = _merge_sharded_checkpoints(
-                        sharded_ckpt_cached_folder, sharded_metadata, dduf_entries=dduf_entries
+                model = cls.from_config(config, **unused_kwargs)
+
+                # Convert the weights
+                from .modeling_pytorch_flax_utils import load_flax_checkpoint_in_pytorch_model
+
+                model = load_flax_checkpoint_in_pytorch_model(model, resolved_archive_file)
+            else:
+                # in the case it is sharded, we have already the index
+                if is_sharded:
+                    resolved_archive_file, sharded_metadata = _get_checkpoint_shard_files(
+                        pretrained_model_name_or_path,
+                        index_file,
+                        cache_dir=cache_dir,
+                        proxies=proxies,
+                        local_files_only=local_files_only,
+                        token=token,
+                        user_agent=user_agent,
+                        revision=revision,
+                        subfolder=subfolder or "",
+                        dduf_entries=dduf_entries,
                     )
-                    logger.info("Merged sharded checkpoints as `hf_quantizer` is not None.")
-                    is_sharded = False
+                elif use_safetensors:
+                    try:
+                        resolved_archive_file = _get_model_file(
+                            pretrained_model_name_or_path,
+                            weights_name=_add_variant(SAFETENSORS_WEIGHTS_NAME, variant),
+                            cache_dir=cache_dir,
+                            force_download=force_download,
+                            proxies=proxies,
+                            local_files_only=local_files_only,
+                            token=token,
+                            revision=revision,
+                            subfolder=subfolder,
+                            user_agent=user_agent,
+                            commit_hash=commit_hash,
+                            dduf_entries=dduf_entries,
+                        )
 
-            elif use_safetensors and not is_sharded:
-                try:
-                    model_file = _get_model_file(
+                    except IOError as e:
+                        logger.error(f"An error occurred while trying to fetch {pretrained_model_name_or_path}: {e}")
+                        if not allow_pickle:
+                            raise
+                        logger.warning(
+                            "Defaulting to unsafe serialization. Pass `allow_pickle=False` to raise an error instead."
+                        )
+
+                if resolved_archive_file is None:
```

Comment:
```suggestion
                if resolved_archive_file is None and not is_sharded:
```
not sure, but it was like this before and shared checkpoints use `_get_checkpoint_shard_files`, no?

Reply:
Yeah, we can be extra cautious with that but `resolved_archive_file` should be different from `None` after  `_get_checkpoint_shard_files`. Otherwise, it will trigger an error in that function.